x-pack/test/functional/services/transform/date_picker.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export function TransformDatePickerProvider({ getService, getPageObjects }: FtrProviderContext) {
+  const find = getService('find');
+  const testSubjects = getService('testSubjects');
+  const pageObjects = getPageObjects(['timePicker']);
+
+  return {
+    async assertSuperDatePickerToggleQuickMenuButtonExists() {
+      await testSubjects.existOrFail('superDatePickerToggleQuickMenuButton');
+    },
+
+    async openSuperDatePicker() {
+      await testSubjects.click('superDatePickerToggleQuickMenuButton');
+      await testSubjects.existOrFail('superDatePickerQuickMenu');
+    },
+
+    async quickSelect(timeValue: number = 15, timeUnit: string = 'y') {
+      const quickMenuElement = await testSubjects.find('superDatePickerQuickMenu');
```

Comment:
Nit: Since these two lines are repeated throughout,

```
          await transform.datePicker.openSuperDatePicker();
          await transform.datePicker.quickSelect(15, 'm');
```
Would it be better to add that to this `quickSelect`?
```suggestion
      await this.openSuperDatePicker();
      const quickMenuElement = await testSubjects.find('superDatePickerQuickMenu');
```

Reply:
Updated in [6610cd0](https://github.com/elastic/kibana/pull/149049/commits/6610cd00ed53fb5516b3ef5f90fe58c3b9c4c749).